onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -892,7 +958,7 @@ common::Status TensorrtExecutionProvider::Provider_Compile(const std::vector<onn
       const std::unordered_map<std::string, int>& output_indexes = (trt_state->output_info)[0];
       const std::unordered_map<std::string, int>& output_types = (trt_state->output_info)[1];
       auto& shape_ranges = trt_state->input_shape_ranges;
-      auto trt_builder = trt_state->builder;
+      auto trt_builder = trt_state->builder->get();
```

Comment:
I hope we can clean up our usage of smart pointers.
e.g. this section of code where we are using get() to get raw pointers for builder, engine, context.
Glancing at the code, I don't think we really need raw pointers. it is difficult to understand ownership.
e.g. lines 982, 983 are calling reset() manually 
it would be better if we kept things as smart pointer and use std::move() to transfer ownership where appropriate.

Reply:
we can consider refactoring this in a subsequent PR.

Reply:
trt_state->builder is a pointer to unique_pointer. It's a bit confusing here. In next PR I will refactor it to get a more clean solution.